iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/metadata/TableMetadataImpl.java

```diff
@@ -90,7 +88,14 @@ public boolean tableExists(QualifiedObjectName name) {
 
   @Override
   public Optional<TableSchema> getTableSchema(SessionInfo session, QualifiedObjectName name) {
-    TsTable table = tableCache.getTable(name.getDatabaseName(), name.getObjectName());
+    String databaseName = name.getDatabaseName();
+    String tableName = name.getObjectName();
+    // get TableSchema of internal table
+    if (databaseName.equals(INFORMATION_SCHEMA)) {
+      return InformationSchemaTable.getTableSchemaFromStringValue(tableName);
+    }
+
+    TsTable table = tableCache.getTable(databaseName, tableName);
```

Comment:
Actually you don't need to do this, @Caideyipi will take over it and will handle it in `tableCache`.
@Caideyipi can remove this code block in your implementation